detection_rules/packaging.py

```diff
@@ -498,7 +498,8 @@ def _generate_registry_package(self, save_dir):
         # shutil.copyfile(CHANGELOG_FILE, str(rules_dir.joinpath('CHANGELOG.json')))
 
         for rule in self.rules:
-            rule.save_json(Path(rules_dir.joinpath(f'rule-{rule.id}.json')))
+            with Path(rules_dir.joinpath(f'rule-{rule.id}.json')).open("w", encoding="utf-8") as f:
+                json.dump(rule.get_asset(), f, indent=2, sort_keys=True)
```

Comment:
```suggestion
            rules_dir.joinpath(f'rule-{rule.id}.json').write_text(
                json.dumps(rule.get_asset(), indent=2, sort_keys=True), encoding="utf-8")
```

Reply:
ytho

Reply:
because we stated that a goal was to defer as much file read and writes to `Path` that made sense, for consistency

Reply:
still uses Path, but also `with` vs `write_text`